subprojects/core/src/main/java/org/gradle/internal/file/JarCache.java

```diff
@@ -34,15 +33,10 @@ public JarCache(FileHasher fileHasher) {
 
     /**
      * Returns a cached copy of the given file. The cached copy is guaranteed to not be modified or removed.
-     *
-     * @param original The source file.
-     * @param baseDirFactory A factory that can provide a base directory for the file cache.
-     * @return The cached file.
      */
-    public File getCachedJar(File original, Factory<File> baseDirFactory) {
```

Comment:
Any idea why a factory has been used in the first place?

Reply:
Nope. Maybe at some point it was expected that the cache dir would change, and was kept lazy?